### ISSUE TITLE
Websocket onclose function is not called in Android

### ISSUE THREAD

**Issue by tli4** [facebook/react-native#18775]
- [x] I have reviewed the [documentation](https://facebook.github.io/react-native)
- [x] I have searched [existing issues](https://github.com/facebook/react-native/issues)
- [x] I am using the [latest React Native version](https://github.com/facebook/react-native/releases)

Websocket connection closed by server will not trigger onclose callback.

## Environment
Environment:
  OS: macOS High Sierra 10.13.3
  Node: 7.10.1
  Yarn: Not Found
  npm: 5.7.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 2.3 AI-162.3871768

Packages: (wanted => installed)
  react: 16.2.0 => 16.2.0
  react-native: ^0.53.0 => 0.53.0

For websocket server-side, I use [ratchetPHP](https://github.com/ratchetphp/Ratchet)

## Steps to Reproduce
<pre>
connect = () => {
    this.ws = new WebSocket('URL')
    this.ws.onmessage = (e) => {
      json = JSON.parse(e.data)
      console.log(json)
    }

    this.ws.onclose = (e) => {
      console.log('closed')
    }
}

componentWillReceiveProps(nextProps){
  this.connect()
}
</pre>

Then close the connection on server.

## Expected Behavior
onclose function should be called.

## Actual Behavior
When building in Android,
close request send from client will trigger the onclose function,
but RN can't detect a connection closed by server.

**Comment by tli4** [comment]
Now I update to react-native 0.55.0
<pre>
Environment:
  OS: macOS High Sierra 10.13.3
  Node: 7.10.1
  Yarn: Not Found
  npm: 5.7.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 2.3 AI-162.3871768

Packages: (wanted => installed)
  react: ^16.3.1 => 16.3.1
  react-native: ^0.55.0 => 0.55.0
</pre>

The same issue is still happening. 

If I close the connection from server, the onclose is not called.
If I close the connection from client, if breaks on:
WebSocketModule.close got 1 arguments, expected 3.

which is caused by [b9be289](https://github.com/facebook/react-native/commit/b9be289)

**Comment by tracinah** [comment]
👌

**Comment by westcode** [comment]
This issue is still there @ latest version.

**Comment by cristiantx** [comment]
Using 0.56.0 with the same issue.
Is there any workaround to know when the WebSocket closes?

**Comment by mmazzarolo** [comment]
I can confirm the issue is still here on the latest version of RN.

I was able to fix it by adding the following listener to `react-native/ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java`:  
```java
@Override
public void onClosing(WebSocket webSocket, int code, String reason) {
    webSocket.close(code, reason);
}
```

**Comment by wuyue1019** [comment]
Still the same issue on the latest RN 0.57.7. Confirmed the onClosing fix code is there, but no luck.
This issue has blocked my RN upgrading since 0.44.3 -_-!
Did I miss anything?

  React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.6
      CPU: (4) x64 Intel(R) Core(TM) i5-6600 CPU @ 3.30GHz
      Memory: 2.23 GB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 8.4.0 - /usr/local/bin/node
      Yarn: 1.3.2 - /usr/local/bin/yarn
      npm: 5.3.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
      Android SDK:
        API Levels: 23, 24, 26, 27
        Build Tools: 23.0.1, 23.0.3, 26.0.1, 27.0.3
        System Images: a... | Intel x86 Atom_64, a...gle_apis | Google APIs Intel x86 Atom_64 ..., a... | Intel x86 Atom_64, a...google_apis | Google APIs Intel x86 Atom Sys...
    IDEs:
      Android Studio: 3.1 AI-173.4907809
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.1 => 16.6.1 
      react-native: 0.57.7 => 0.57.7 
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7

**Comment by wuyue1019** [comment]
I guess to be more accurate, when the server side closed the connection, my app got onerror first after a few seconds' hung-up and then the onclose message. But since onerror came in first, my app would start to try a reconnection ... . Is onerror expected first in this case? iPhone works fine.

**Comment by Salakar** [comment]
👋 hey all, as mentioned by @PeterKruzlics above, this was fixed in v0.57.6 so I'll go ahead and close this now. 

@wuyue1019 that sounds like a different issue to the original issue specified here. If this is still a concern for you please could you raise a new detailed issued with a reproducible example if possible so it can be looked at?

Thank you
